staramr/detection/AMRDetection.py

```diff
@@ -122,6 +131,59 @@ def run_amr_detection(self, files, pid_threshold, plength_threshold_resfinder, p
         self._detailed_summary_dataframe = self._create_detailed_amr_summary(files, self._resfinder_dataframe,
                                                                              self._pointfinder_dataframe,
                                                                              self._plasmidfinder_dataframe)
+                                                                             
+    def _validate_files(self, files: List[str], ignore_invalid_files: bool) -> List[str]:
+        total_files = len(files)
+        removeable_files = []
+
+        for file in files:
+            # Check if the file is not a directory
+            if os.path.isdir(file):
+                if ignore_invalid_files:
+                    logger.warning('--ignore-invalid-files is set, skipping directory {}'.format(file))
+                    removeable_files.append(file)
+                else:
+                    raise Exception('Directory {} is invalid, please use --ignore-invalid-files to skip over this directory'.format(file))
+            else:
+                # Will raise an error if the input returns an empty generator on non-FASTA files, returns a boolean
+                validInput = any(SeqIO.parse(file, "fasta"))
+
+                if not validInput:
+                    if ignore_invalid_files:
+                            logger.warning('--ignore-invalid-files is set, skipping file {}'.format(file))
+                            removeable_files.append(file)
+                    else:
+                        raise Exception('File {} is invalid, please use --ignore-invalid-files to skip over invalid input files'.format(file))
+            
+        # Check to see if the invalid file is not the only file in the directory
+        if total_files == len(removeable_files):
+            raise Exception('Cannot produce output due to no valid input files')
+
+        # Remove the skipped files
+        if ignore_invalid_files:
+            for file in removeable_files:
+                files.remove(file)
+
+        # Check if there are any duplicate sequence id's in the valid files
+        for file in files:
```

Comment:
You could likely merge this bit of code into the for loop above. That way you're only looping over the files and reading them once, instead of twice.

Reply:
Done, I refactored it to check for duplicates after the code knows it's valid